cmd/internal/types.go

```diff
@@ -148,6 +151,90 @@ func QueryResultToRecords(qr *sqltypes.Result) []map[string]interface{} {
 	return data
 }
 
+// After the initial COPY phase, enum and set values may appear as an index instead of a value.
+// For example, a value might look like a "1" instead of "apple" in an enum('apple','banana','orange') column)
+func parseValue(val sqltypes.Value, columnType string) sqltypes.Value {
+	if strings.HasPrefix(columnType, "enum") {
+		values := parseEnumOrSetValues(columnType)
+		return mapEnumValue(val, values)
+	} else if strings.HasPrefix(columnType, "set") {
+		values := parseEnumOrSetValues(columnType)
+		return mapSetValue(val, values)
+	}
+
+	return val
+}
+
+// Takes enum or set column type like ENUM('a','b','c') or SET('a','b','c')
+// and returns a slice of values []string{'a', 'b', 'c'}
+func parseEnumOrSetValues(columnType string) []string {
```

Comment:
Putting the "parsing" logic here does result in an extra parse on every enum/set column value. If this is too much processing, we could "save" the parsed values somewhere upstream - however, that would require refactoring a great deal more of this code since, as it is currently, we don't pass a lot of metadata around during the `READ` phase.